cmd/wampproto/main.go

```diff
@@ -1,20 +1,92 @@
 package main
 
 import (
-	"encoding/hex"
+	"fmt"
 	"log"
+	"os"
 
-	"github.com/xconnio/wampproto-go/messages"
-	"github.com/xconnio/wampproto-go/serializers"
+	"github.com/alecthomas/kingpin/v2"
+
+	"github.com/xconnio/wampproto-cli"
+	"github.com/xconnio/wampproto-go/auth"
 )
 
-func main() {
-	hello := messages.NewHello("realm1", "test", nil, nil, []string{"anonymous"})
-	serializer := &serializers.MsgPackSerializer{}
-	data, err := serializer.Serialize(hello)
+const versionString = "0.1.0"
+
+const (
+	HexFormat    = "hex"
+	Base64Format = "base64"
+)
+
+type cmd struct {
+	parsedCommand string
+
+	output *string
+
+	auth              *kingpin.CmdClause
+	cryptosign        *kingpin.CmdClause
+	generateChallenge *kingpin.CmdClause
+}
+
+func parseCmd(args []string) (*cmd, error) {
+	app := kingpin.New(args[0], "A tool for testing interoperability between different wampproto implementations.")
+	app.Version(versionString).VersionFlag.Short('v')
+
+	authCommand := app.Command("auth", "Authentication commands.")
+
+	cryptoSignCommand := authCommand.Command("cryptosign", "Commands for cryptosign authentication.")
+
+	c := &cmd{
+		output: app.Flag("output", "Format of the output.").Default("hex").Enum(HexFormat, Base64Format),
+
+		auth: authCommand,
+
+		cryptosign: cryptoSignCommand,
+
+		generateChallenge: cryptoSignCommand.Command("generate-challenge", "Generate a cryptosign challenge."),
+	}
+
+	parsedCommand, err := app.Parse(args[1:])
+	if err != nil {
+		return nil, err
+	}
+	c.parsedCommand = parsedCommand
+
+	return c, nil
+}
+
+func Run(args []string) error {
```

Comment:
```suggestion
func Run(args []string) error {
```
This function may just return the output string and the error. Then it becomes super easy to work in tests as well. No hacks needed to check shell output